ttnn/cpp/ttnn/operations/matmul/matmul.cpp

```diff
@@ -92,6 +92,79 @@ ttnn::Tensor bound_matmul(
     return output_tensor;
 }
 
+Tensor MatmulOperation::invoke(
+    const Tensor& input_tensor_a,
+    const Tensor& input_tensor_b,
+    const bool transpose_a,
+    const bool transpose_b,
+    const MemoryConfig& memory_config,
+    const std::optional<const DataType> dtype,
+    const std::optional<const MatmulProgramConfig> program_config,
+    const std::optional<const std::string>& activation,
+    const std::optional<const DeviceComputeKernelConfig> compute_kernel_config,
+    const std::optional<const CoreGrid> core_grid) {
+    std::optional<CoreCoord> user_core_coord;
+    if (core_grid.has_value()) {
+        user_core_coord = CoreCoord(core_grid->x, core_grid->y);
+    }
+    bool user_run_batched = detail::is_input_batched(input_tensor_b.get_shape());
+    return bound_matmul(
+        input_tensor_a,
+        input_tensor_b,
+        /*bias=*/std::nullopt,
+        Matmul{
+            program_config,
+            /*bcast_batch=*/std::nullopt,
+            memory_config,
+            dtype,
+            compute_kernel_config,
+            /*untilize_out=*/false,
+            user_core_coord,
+            get_fused_activation(activation),
+            user_run_batched,
+            transpose_a,
+            transpose_b},
+        /*queue_id=*/0);
+}
+
+Tensor LinearOperation::invoke(
+    const Tensor& input_tensor_a,
+    const Tensor& input_tensor_b,
+    const std::optional<const Tensor>& bias,
+    const bool transpose_a,
+    const bool transpose_b,
+    const MemoryConfig& memory_config,
+    const std::optional<const DataType> dtype,
+    const std::optional<const MatmulProgramConfig> program_config,
+    const std::optional<const std::string>& activation,
+    const std::optional<const DeviceComputeKernelConfig> compute_kernel_config,
+    const std::optional<const CoreGrid> core_grid) {
+    std::optional<CoreCoord> user_core_coord;
+    if (core_grid.has_value()) {
+        user_core_coord = CoreCoord(core_grid->x, core_grid->y);
+    }
+    bool b_is_batched = detail::is_input_batched(input_tensor_b.get_shape());
+    TT_FATAL(!(b_is_batched && bias.has_value()), "Batched input not supported when bias exists (linear operation).");
+
+    return bound_matmul(
+        input_tensor_a,
+        input_tensor_b,
+        bias,
+        Matmul{
+            program_config,
+            /*bcast_batch=*/std::nullopt,
+            memory_config,
+            dtype,
+            compute_kernel_config,
+            /*untilize_out=*/false,
+            user_core_coord,
+            get_fused_activation(activation),
+            /*user_run_batched=*/false,
+            transpose_a,
+            transpose_b},
+        /*queue_id=*/0);
```

Comment:
ttnn::DefaultQueueId can be taken from `ttnn/common/constants.hpp`

Reply:
I'll deal with this in the next PR. Looking at https://docs.tenstorrent.com/tt-metal/latest/ttnn/ttnn/adding_new_ttnn_operation.html it looks like I should be able to remove queue_id once I set up the ttnn::prim code.